### PR TITLE
feat: Add `--context` flag to choose specific context from kubeconfig

### LINKED ISSUE #925: add support for --context flag like kubectl
### Feature request

`kubectl` has support for a `--context` flag that allows you to specify which cluster in the Kubernetes config file to use. W/o this it makes it harder to switch between kube cluster easily.
For example, I often have Makefiles that point to different clusters and I can just add the `--context` flag to each `kubectl` command w/o having to set KUBECONFIG or use some other tool to set my Kube context before running `make` - which might impact other things going on in my env that point to other clusters.

/kind proposal

### PR DESCRIPTION
**Pull request by dsimansk**
## Description

I've noticed new comments on the issue and though it should be pretty straightforward and great addition to `kn`. Most of the changes are docs (in separate commit).

## Changes

* Add `--context` flag to choose specific context from kubeconfig

## Reference

Fixes #925

### CODE REVIEW

**Comment by rhuss** [comment]
Let try to get this in for 0.21, too.

File: pkg/kn/commands/types.go [hunk 1]
```diff
@@ -183,8 +188,8 @@ func (params *KnParams) GetClientConfig() (clientcmd.ClientConfig, error) {
 
 	paths := filepath.SplitList(params.KubeCfgPath)
 	if len(paths) > 1 {
-		return nil, fmt.Errorf("Can not find config file. '%s' looks like a path. "+
+		return nil, fmt.Errorf("can not find config file. '%s' looks like a path. "+
```
Comment (dsimansk) [inline pkg/kn/commands/types.go:191]: Error messages are changed due to lint warning "Error string should not be capitalized or end with punctuation".

**Comment by dsimansk** [comment]
/retest
```
 Error from server (InternalError): error when creating "https://storage.googleapis.com/knative-releases/eventing/previous/v0.20.0/eventing.yaml": Internal error occurred: failed calling webhook "config.webhook.eventing.knative.dev": Post https://eventing-webhook.knative-eventing.svc:443/config-validation?timeout=2s: no endpoints available for service "eventing-webhook"
Waiting until all pods in namespace knative-eventing are up......................................................................................................................................................

```

**Review by navidshaikh — APPROVED** [review]
/lgtm
/assign @rhuss 
/hold

**Review by rhuss — COMMENTED** [review]
Looks good, I wonder whether we should also add `--cluster` like for kubectl in one go ?

Or should this be better done in a separate PR ?

File: pkg/kn/commands/types.go [hunk 2]
```diff
@@ -167,14 +168,18 @@ func (params *KnParams) RestConfig() (*rest.Config, error) {
 // GetClientConfig gets ClientConfig from KubeCfgPath
 func (params *KnParams) GetClientConfig() (clientcmd.ClientConfig, error) {
 	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	if len(params.KubeContext) > 0 {
```
Comment (rhuss) [inline pkg/kn/commands/types.go:172]:
```suggestion
	if params.KubeContext != "" {
```

**Comment by dsimansk** [comment]
> Looks good, I wonder whether we should also add `--cluster` like for kubectl in one go ?
> 
> Or should this be better done in a separate PR ?

Well, probably makes sense to add it together. I'll take a look right now.

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -17,9 +17,14 @@
 |===
 | | Description | PR
 
+| 🎁
+| Add `--context` and `--cluster` flags to choose specific context from kubeconfig
```
Comment (dsimansk) [inline CHANGELOG.adoc:21]: I couldn't decide if that shouldn't be split to 2 entries per flag. Please let me know wdyt.

**Comment by dsimansk** [comment]
Infra issue
/retest

File: CHANGELOG.adoc (hunk 3 above)
Comment (rhuss) [reply CHANGELOG.adoc:21]: I think this is fine

**Review by rhuss — APPROVED** [review]
thanks !

/lgtm
/unhold
/approve

**Pull request merged**